From FEDORA_PATCHES Mon Sep 17 00:00:00 2001
From: Fedora GDB patches <invalid@email.com>
Date: Fri, 27 Oct 2017 21:07:50 +0200
Subject: gdb-moribund-utrace-workaround.patch

FileName: gdb-moribund-utrace-workaround.patch

;; Workaround non-stop moribund locations exploited by kernel utrace (BZ 590623).
;;=push+jan: Currently it is still not fully safe.

https://bugzilla.redhat.com/show_bug.cgi?id=590623
http://sources.redhat.com/bugzilla/show_bug.cgi?id=11593

Bug in FSF GDB exploited by the ptrace-on-utrace interaction.
---
 gdb/breakpoint.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gdb/breakpoint.c b/gdb/breakpoint.c
index 269928a794..fb129bf38a 100644
--- a/gdb/breakpoint.c
+++ b/gdb/breakpoint.c
@@ -12105,6 +12105,8 @@ update_global_location_list (enum ugll_insert_mode insert_mode)
 		 traps we can no longer explain.  */
 
 	      old_loc->events_till_retirement = 3 * (thread_count () + 1);
+	      /* Red Hat Bug 590623.  */
+	      old_loc->events_till_retirement *= 10;
 	      old_loc->owner = NULL;
 
 	      VEC_safe_push (bp_location_p, moribund_locations, old_loc);
-- 
2.14.3

